runtime/expr/function/grep.go

```diff
@@ -0,0 +1,43 @@
+package function
+
+import (
+	"fmt"
+
+	"github.com/brimdata/zed"
+	"github.com/brimdata/zed/runtime/expr"
+	"golang.org/x/text/unicode/norm"
+)
+
+type Grep struct {
+	grep   expr.Evaluator
+	grestr string
```

Comment:
Nit: I think `pattern` would be a better name.
```suggestion
	pattern string
```